tools/static_word_checks.py

```diff
@@ -0,0 +1,160 @@
+import argparse
+import os
+import re
+import sys
+
+
+class StaticChecker:
+    """Run simple checks on the entire document or specific lines."""
+
+    double_regex = re.compile(r"(\s)/(u|r)/")
+
+    def __init__(self, replace: bool):
+        """Instantiates the class.
+
+        :param replace: Whether or not to make replacements.
+        """
+        self.full_file_checks = [
+            self.check_for_code_statement,
+            self.check_for_double_syntax
+            # add more checks to the list as they are added
+        ]
+        self.line_checks = [
+            self.check_for_noreturn,
+            # add more checks to the list as they are added
+        ]
+        self.replace = replace
+
+    def check_for_code_statement(self, filename: str, content: str) -> bool:
+        """Checks the code for ``.. code::`` statements.
+
+        :param filename: The name of the file to check & replace.
+        :param content: The content of the file
+        :returns: A boolean with the status of the check
+        """
+        if ".. code::" in content:
+            newcontent = content.replace(".. code::", ".. code-block::")
+            if self.replace:
+                with open(filename, "w") as fp:
+                    fp.write(newcontent)
+                print(
+                    "{filename}: Replaced all ``code::`` to "
+                    "``code-block::``".format(filename=filename)
+                )
+                return True
+            print(
+                "{filename}; This file uses the `code::` syntax, please change"
+                " the syntax to ``code-block::``.".format(filename=filename)
+            )
+            return False
+        return True
+
+    def check_for_double_syntax(self, filename: str, content: str) -> bool:
+        """Checks a file for double-slash statements (``/r/`` and ``/u/``).
+
+        :param filename: The name of the file to check & replace.
+        :param content: The content of the file
+        :returns: A boolean with the status of the check
+        """
+        if (
+            os.path.join("praw", "const.py")  # fails due to bytes blocks
+            in filename
+        ):
+            return True
+        match = self.double_regex.search(content)
+        if match is not None:
+            orig_str = match.group(0)
+            newstr = "{}{}/".format(match.group(1), match.group(2))
+            newcontent = content.replace(orig_str, newstr)
```

Comment:
I had to do this because self. double_regex.sub replaced the entire file with `\1\2/` and not the captured groups.